main.cpp

```diff
@@ -1,54 +1,87 @@
 #include "nwpwin.h"
 #include "res.h"
 
-// TODO: prepare classes (edit, button, list_box) for child windows
-// TODO: derive from window, override class_name
+
+class list_box : public vsite::nwp::window {
+	std::string class_name() override { return "LISTBOX"; };
+};
+class edit : public vsite::nwp::window {
+	std::string class_name() override { return "EDIT"; };
+};
+class button : public vsite::nwp::window {
+	std::string class_name() override { return "button"; };
+};
 
 class main_window : public vsite::nwp::window
 {
 protected:
 	int on_create(CREATESTRUCT* pcs) override;
 	void on_command(int id) override;
 	void on_destroy() override;
+
+private:
+	edit et;
+	list_box lb;
+	button btn1;
+	button btn2;
 };
 
 int main_window::on_create(CREATESTRUCT* pcs)
 {
-	// TODO: create all child windows
-	// TODO: disable "Remove" button
+
+	lb.create(*this, WS_CHILD | WS_VISIBLE | WS_BORDER, 0, IDC_LB, 10, 5, 100, 100);
+	et.create(*this, WS_CHILD | WS_VISIBLE | WS_BORDER, 0, IDC_EDIT, 130, 5, 100, 25);
+	btn1.create(*this, WS_CHILD | WS_VISIBLE | WS_BORDER, "Add", IDC_ADD, 130, 40, 100, 25);
+	btn2.create(*this, WS_CHILD | WS_VISIBLE | WS_BORDER, "Remove", IDC_REMOVE, 130, 75, 100, 25);
+
+	EnableWindow(btn2, FALSE);
 	return 0;
 }
 
-void main_window::on_command(int id){
-	switch(id){
-		case ID_FILE_EXIT:
-			// TODO: close main window
-			break;
-		case ID_HELP_ABOUT:
-			// TODO: show dialog with text
-			break;
-		case IDC_ADD:
-			// TODO: get text from edit control
-			// TODO: add string to listbox control
-			// TODO: enable "Remove" button
-			break;
-		case IDC_REMOVE:
-			// TODO: get listbox selection
-			// TODO: if there is a selection, delete selected string
-			// TODO: disable "Remove" button if listbox is empty
-			break;
+void main_window::on_command(int id) {
+	switch (id) {
+	case ID_FILE_EXIT:
+
+		PostQuitMessage(0);
+		break;
+	case ID_HELP_ABOUT:
+
+		MessageBox(*this, TEXT("List"), TEXT("About"), MB_OK | MB_ICONINFORMATION);
+		break;
+	case IDC_ADD:
+
+		char text[36];
+		GetDlgItemText(*this, IDC_EDIT, text, 20);
+
+		SendDlgItemMessage(*this, IDC_LB, LB_ADDSTRING, 0, (LPARAM)text);

```

Comment:
Zašto ovdje `SendDlgItemMessage` (sa _id_-em _listbox_-a), a na _Remove_ `SendMessage` memberu `lb`?
(i jedno i drugo je OK, samo mi nije jasno)

Reply:
Inkonzistentnosti nisu namjerne, vjerojatno sam zanemario kod iz IDC_ADD case-a dok sam radio IDC_REMOVE case

Reply:
Jeste li uopće probali nakon zadnjih promjena? Zamijenili ste i `LB_GETCURSEL` i `LB_DELETESTRING` i `LB_GETCOUNT` sa `LB_ADDSTRING` (?!)